From 8bd7e076bc3ad047cdf814d1f003e41fc78fef78 Mon Sep 17 00:00:00 2001
From: H. Peter Anvin <hpa@zytor.com>
Date: Thu, 15 Apr 2010 13:50:27 +0800
Subject: [JOBS] Fix for job control off warning

There seems to be a problem with the new version of dash
with job control off.  I can't tell if it is just a warning or is a
manifest bug.

usr/dash/trap.c: In function `exitshell':
usr/dash/trap.c:376: warning: suggest braces around empty body in an `if' statement

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit dc2bc17c7bbe3188cfb0c695dc472b39bcf0b27a)

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 ChangeLog  |    4 ++++
 src/jobs.h |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index a3c1242..cc47e28 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2010-04-15  H. Peter Anvin <hpa@zytor.com>
+
+	* Fix for job control off warning.
+
 2010-04-02  Herbert Xu <herbert@gondor.apana.org.au>
 
 	* Use faccessat if available.
diff --git a/src/jobs.h b/src/jobs.h
index 26e421d..9c095ea 100644
--- a/src/jobs.h
+++ b/src/jobs.h
@@ -105,5 +105,5 @@ int waitforjob(struct job *);
 int stoppedjobs(void);
 
 #if ! JOBS
-#define setjobctl(on)	/* do nothing */
+#define setjobctl(on) ((void)(on))	/* do nothing */
 #endif
-- 
1.7.4.1

